### PR TITLE
Generalize NFE bans into validator rule

### PR DESCRIPTION
**Pull request by TheMezStrikes**
so various NFE tournaments like old generations or Mix & Mega can be played without gentleman's rule or complex banlists. Tried to ask @TheImmortal for permission beforehand but didn't get an answer so figured I'd just propose it and see if it gets approved

### CODE REVIEW

File: data/rulesets.js [hunk 1]
```diff
@@ -535,6 +535,20 @@ let BattleFormats = {
 			this.add('rule', 'Z-Move Clause: Z-Moves are banned');
 		},
 	},
+	nfeclause: {
+		effectType: 'ValidatorRule',
+		name: 'NFE Clause',
+		desc: "Bans Pok&eacute;mon that cannot evolve",
```
Comment (Zarel) [inline data/rulesets.js]: I feel like "cannot evolve" is a bit vague. I would say either "Bans Pokemon that are fully evolved", or "Bans Pokemon that are fully evolved or can't evolve".

Comment (TheImmortal) [reply data/rulesets.js]: Agree. Should be the latter.

Comment (TheMezStrikes) [reply data/rulesets.js]: Should be good now

**Comment by Zarel** [comment]
I'm not super sold on the name, because clause names are usually what's banned, not what's not banned, but whatever.

**Pull request merged**